integrations/appboy/lib/index.js

```diff
@@ -373,30 +373,40 @@ Appboy.prototype.track = function(track) {
   var properties = track.properties();
   // remove reserved keys from custom event properties
   // https://www.appboy.com/documentation/Platform_Wide/#reserved-keys
-  var reserved = [
-    'time',
-    'product_id',
-    'quantity',
-    'event_name',
-    'price',
-    'currency'
-  ];
-  each(function(key) {
-    delete properties[key];
-  }, reserved);
-
-  // Remove nested objects as Braze doesn't support objects in tracking calls
-  // https://segment.com/docs/destinations/braze/#track
-  each(function(value, key) {
-    if (value != null && typeof value === 'object') {
-      delete properties[key];
-    }
-  }, properties);
 
   if (userId) {
     window.appboy.changeUser(userId);
   }
-  window.appboy.logCustomEvent(eventName, properties);
+  if (properties.revenue) {
+    window.appboy.logPurchase(
```

Comment:
@pankaj-webonise seems this needs to be changed to support product array.
`track.products()` gives multiple products and we need to iterate over it.